ecosystem/typescript/sdk_v2/src/types/index.ts

```diff
@@ -155,6 +155,10 @@ export type PendingTransactionResponse = {
   signature?: TransactionSignature;
 };
 
+export function isPendingTransaction(txn: TransactionResponse): txn is PendingTransactionResponse {
+  return txn.type === "pending_transaction";
```

Comment:
We should make an enum for this and document where it comes from. Can just copy the values from rust, I linked it below :D

https://github.com/aptos-labs/aptos-core/blob/3e1f8dd9c508d5f241aba48a03fe76cf2e934d61/api/types/src/transaction.rs#L219